lib/raven.interceptor.ts

```diff
@@ -1,63 +1,79 @@
+  Injectable,
   NestInterceptor,
   CallHandler,
 } from '@nestjs/common';
 import { IRavenInterceptorOptions } from './raven.interfaces';
 import { Observable } from 'rxjs';
 import { tap } from 'rxjs/operators';
-import * as Sentry from '@sentry/minimal'
+import * as Sentry from '@sentry/minimal';
 import { Scope } from '@sentry/hub';
-import { RpcArgumentsHost, WsArgumentsHost, HttpArgumentsHost } from '@nestjs/common/interfaces';
-import { Handlers } from '@sentry/node'
-import { GqlArgumentsHost, GraphQLArgumentsHost } from '@nestjs/graphql'
+import {
+  RpcArgumentsHost,
+  WsArgumentsHost,
+  HttpArgumentsHost,
+} from '@nestjs/common/interfaces';
+import { Handlers } from '@sentry/node';
+import { GqlArgumentsHost, GraphQLArgumentsHost } from '@nestjs/graphql';
 
 @Injectable()
 export class RavenInterceptor implements NestInterceptor {
-  
-  constructor(
-    private readonly options: IRavenInterceptorOptions = { context: 'Http' },
-  ) {
-  }
-  
-  intercept(
-    context: ExecutionContext,
-    next: CallHandler,
-  ): Observable<any> {
-    const http = context.switchToHttp();
-    const ws = context.switchToWs();
-    const rpc = context.switchToRpc();
-
-    // Default to Http as conteext
-    if (!this.options.context) this.options.context = 'Http';
+  constructor(private readonly options: IRavenInterceptorOptions = {}) {}
 
+  intercept(context: ExecutionContext, next: CallHandler): Observable<any> {
     // first param would be for events, second is for errors
-    return next.handle()
-    .pipe(
-      tap(null, (exception) => {
+    return next.handle().pipe(
+      tap(null, exception => {
         if (this.shouldReport(exception)) {
           Sentry.withScope(scope => {
-            // TODO: When https://github.com/nestjs/nest/issues/1581 gets implemented switch to that
-            switch(this.options.context){
-              case 'Http': return this.captureHttpException(scope as any, http, exception);
-              case 'Ws': return this.captureWsException(scope as any, ws, exception);
-              case 'Rpc': return this.captureRpcException(scope as any, rpc, exception);
-              case 'GraphQL': return this.captureGraphQLException(scope as any, GqlArgumentsHost.create(context), exception);
+            switch (context.getType()) {
+              case 'http':
+                return this.captureHttpException(
+                  scope as any,
+                  context.switchToHttp(),
+                  exception,
+                );
+              case 'ws':
+                return this.captureWsException(
+                  scope as any,
+                  context.switchToWs(),
+                  exception,
+                );
+              case 'rpc':
+                return this.captureRpcException(
+                  scope as any,
+                  context.switchToRpc(),
+                  exception,
+                );
+              default:
+                return this.captureGraphQLException(
```

Comment:
`graphql` is not include default ContextType [(refs)](https://github.com/nestjs/graphql/blob/b17ab3b8e327ad55b5f66178547de64826ae0f6e/lib/services/gql-execution-context.ts#L5).

making the default `graphql` can be dangerous, so make the default optional if necessary.

```suggestion
              if (!this.options.withGraphql) return this.captureException(scope, exception);
                return this.captureGraphQLException(
```

Reply:
I wouldn't go with the default, and also try to avoid adding `withGraphql` option.

Are there any other contexts that are added by other modules?

Reply:
Fixed :)